server/app.js

```diff
@@ -476,12 +533,15 @@ app.get('/vault', asyncHandler(async(req, res, next)=>{
 }));
 
 //Send rendered page
-app.use(asyncHandler(async (req, res, next)=>{
-	if (!req.route) return res.redirect('/'); // Catch-all for invalid routes
-		
-	const page = await renderPage(req, res);
-	if(!page) return;
-	res.send(page);
+app.use(asyncHandler(async (req, res, next) => {
+    if (!req.route && !req.path.startsWith('/api/')) {
+        return res.redirect('/');
+    }
```

Comment:
Well, i added the change to stop my call rerouting in a loop, but i can test this again now that it is finished and see if i really need it. I'd keep it because any error in the api would end here, and this patch makes an error like that easier to identify. Not sure.

Reply:
Any error in the API should be caught in the API and send the  error response to the client. If the API call is still reaching this point that is a bug.

Reply:
change reverted